### PR TITLE
More methods

### PR DESCRIPTION
**Pull request by daedalus28**
- Add `renamePropertyOn`, `unsetProperty`
- Add `updateOwn`, `updateOwnOn`, `updateSome`, `updateSomeOn`, `updateAll`, `updateAllOn`
- Add `crunchWhitespace`
- Export `writeTreeNode`
- Add `chunkByValue`
- Pass indexes to `arrayToObject`
- Add `omitByIndexed` conversion

Also begin adding `@since` and `@type` in jsdoc (preparing for proper TS typing:)

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Could not run `duti:fix` command successfully
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Branch being merged does not follow [Git Flow](http://nvie.com/posts/a-successful-git-branching-model/)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>You should add at least 1 more reviewer to the PR</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 788af424c27ff496f686afa9eb72d133360bb9a6
</p>

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+# 1.75.0
+
+- Add `renamePropertyOn`, `unsetProperty`
```
Comment (daedalus28) [inline CHANGELOG.md:3]: `unsetProperty` -> `popProperty`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+- Add `updateOwn`, `updateOwnOn`, `updateSome`, `updateSomeOn`, `updateAll`, `updateAllOn`
```
Comment (daedalus28) [inline CHANGELOG.md:4]: `updateOwn` -> `updateIfExists` 
`updateOwnOn` -> `updateOnIfExists` 

`updateSome` -> `updatePaths`
`updateSomeOn` -> `updateAllOn`
`updateAll` -> `updateAllPaths`
`updateAllOn` -> `updateAllPathsOn`

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,13 @@
...
+- Add `crunchWhitespace`
+- Export `writeTreeNode`
+- Add `chunkByValue`
+- Pass indexes to `arrayToObject`
+- Add `omitByIndexed` conversion
```
Comment (daedalus28) [inline CHANGELOG.md:9]: ADD:

`equalsAt`
`omitByPrefixes`
`isSubset`
`sizeBy`
`matchesBy`
`matchesBySome`
`mapValuesIndexed`

Comment (daedalus28) [reply CHANGELOG.md:9]: Skipping `equalsAt` and `omitByPrefixes` due to low known usage

**Pull request merged**